app/models/king.rb

```diff
@@ -1,12 +1,90 @@
 class King < Piece
-  # This method does not yet consider castling.
   def valid_move?(new_x, new_y)
     return false if off_board?(new_x, new_y)
     return false if current_square?(new_x, new_y)
 
     x_offset, y_offset = movement_by_axis(new_x, new_y)
-    return false unless (-1..1).cover?(x_offset) && (-1..1).cover?(y_offset)
+    return true if (-1..1).cover?(x_offset) && (-1..1).cover?(y_offset)
 
-    true
+    valid_castling?(new_x, new_y)
+  end
+
+  # An addition to the move_to! method found within the Piece model.
+  # This determines which rook is to be castled and moves it to the
+  # appropriate square.
+  def move_to!(destination_x, destination_y)
+    return false unless valid_move?(destination_x, destination_y)
+    castling = (x_coord - destination_x).abs == 2
+
+    # Call the move_to! method within the Piece model and move the king.
+    super
+
+    # If all of the requirements are met for valid_move? AND the intention is to castle;
+    # place the rook in its appropriate position to complete the castling move.
+    if castling
+      rook_origin_file = (destination_x == 3) ? 1 : 8
+      rook_destination_file = (destination_x == 3) ? 4 : 6
+      castled_rook = game.piece_at(rook_origin_file, destination_y)
+      castled_rook.update_attributes(x_coord: rook_destination_file,
+                                     y_coord: destination_y,
+                                     moved: true)
+    end
+  end
+
+  private
+
+  # Determine if castling is allowed.
+  def valid_castling?(new_x, new_y)
+    # The right to castle has been lost if the king has moved.
+    return false if moved?
```

Comment:
This is OK, with all these guard clauses, but I think you can leverage exceptions here. Basically you move all these things into assertion methods, so something like:
```
def assert_king_has_not_moved!
  raise KingHasMovedError if moved?
end
```
then this line would just be `assert_king_has_not moved!` here, and you'd have a few of those in a row, one for each.
of these guard clauses. 

In this case, you'd have to put the assertions in a `begin` / `rescue` block and have the rescue case return false.
Does that make sense?

(To be clear, I think it's OK as-is, this just makes things clearer)